## Profitable App Profiles for the App Store and Google Play Market

### Introduction:  
Xletra, a Company which provides App solutions to Users at a free of cost. The Main source of revenue being in-app ads which allows the developers to provide more Apps that can be downloaded and installed free of cost.

Working as a Data Analyst, my main Job is to analyse various Datasets available and in turn come up with suggestions to help the App Developers decide on various exciting App Solutions.  
This will help in attracting more Users and will also help in boosting the revenue.
### Problem Statement: 
There are many different types of App Caegories available in both Apple and GooglePlay Store, deciding on building an App which is free, boosts the user base by being unique and at the same time helps in increasing the revenue of the Company.

### Aim: 
To collect and Analyze Data on the types of Mobile Applications available and those which are likely to attract more users on GooglePlay and the App Store.

### Problem Approach:
1. Seaching for Suitable Datasets
2. Exploring the Data
3. Cleaning the Data
4. Analysing Data
5. Presenting the most viable Solution



### Solution: 
There were around 7198 apps in the Apple Store and around 10842 apps in the GooglePlay Store (In the Sample Dataset). After cleaning the Data, we are left with 3221 in iOS and 8861 in Android to choose from. 
After Analyzing I conluded that Building a Book Store based App, having the Most recent Books along with other Feature like Discussion Forums,etc. will be a beneficial idea. 



### A small insight on the Datasets:
As of September 2018, There were approximately 2 Million iOS Apps available on the App Store and 2.1 Million Android apps on the Google Play.  
There are Two Data sets which are available and will be used in the Project:  
 1. [A data set](https://www.kaggle.com/lava18/google-play-store-apps) containing data about approximately 10,000 Android apps from Google Play; the data was collected in August 2018.  
 2. [A data set](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) containing data about approximately 7,000 iOS apps from the App Store; the data was collected in July 2017

## In-details explanation

### Step 1:  
##### Let's Open the contents of the Datasets and define a function *explore_data* to display few rows of the Datasets (As shown in the code below)

In [1]:
def opening(data_set):
    from csv import reader
    opened_file=open(data_set)
    read_file=reader(opened_file)
    apps_data=list(read_file)
    return apps_data
iOS_Apps=opening('AppleStore.csv') #Apple Store 
Android_Apps=opening('googleplaystore.csv') #GooglePlay Store

In [2]:
def explore_data(dataset,start,end,rows_and_columns=False):
    dataset_slice=dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n')
    if rows_and_columns:
        print('Number of Rows:',len(dataset))
        print('Number of Columns:',len(dataset[0]))
explore_data(iOS_Apps,1,5,True)
print('\n')
print('DataSet of GooglePlay Store:\n')
explore_data(Android_Apps,1,5,True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


Number of Rows: 7198
Number of Columns: 16


DataSet of GooglePlay Store:

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


In [3]:
#Column Names
print ('iOS Apps has the following Columns:'+2*'\n',iOS_Apps[0])
print('\n')
print ('Android Apps has the following Columns:'+2*'\n',Android_Apps[0])

iOS Apps has the following Columns:

 ['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


Android Apps has the following Columns:

 ['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


A detailed explanation of what each Columns stands for can be found in the respective Data Documentation which can be found in the below links :  
   1. [iOS Apps](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps)  
   2. [Google Play Store](https://www.kaggle.com/lava18/google-play-store-apps)  
   

### Step 2 : Data Cleaning - As we deal with Apps which are *Free* to Download and are directed towards the *English-Speaking* audience, We need to first clean the Data of various Languages which might be present and Prepare our Data for the next Process of Analysis

Data Cleaning takes away 80% of a task and the rest 20% is the actual Data Analysis.  
It involves *removing* or *correcting wrong data*, *removing duplicate data* and *modifying the data* to fit the purpose of our analysis. 

On reading the Discussion Forum, we can find the row in which the wrong Data/Column is present


In [4]:
iOS_header=iOS_Apps[0]
GooglePlay_header=Android_Apps[0]
print (len(Android_Apps))
for val in Android_Apps[1:]:
    if len(val)!=len(GooglePlay_header):
        index_value=Android_Apps.index(val)
        print (index_value+1)
        print (Android_Apps[index_value])
        del(Android_Apps[index_value])        

10842
10474
['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In [5]:
print (len(Android_Apps))
for value in iOS_Apps:
    if len(value)!=len(iOS_header):
        print (value[0])

10841


If we skim through the Data set of the Google Play Apps, we can see that there are many Duplicate entries for the same App with little difference. Let's Print few to analyze these duplicate entries.  


In [6]:
duplicates_entries=[]
unique_entries=[]
for duplicates in Android_Apps[1:]:
    if duplicates[0] in unique_entries:
        duplicates_entries.append(duplicates[0])
    else:
        unique_entries.append(duplicates[0])
for valy in Android_Apps[1:]:
    if valy[0]==duplicates_entries[0]:
        print (valy)
        print('\n')
    if valy[0]=='Instagram':
        print(valy)
        print('\n')
print('The No. of duplicate entries is :',len(duplicates_entries))
print('The No. of unique entries is :',len(unique_entries))

['Quick PDF Scanner + OCR FREE', 'BUSINESS', '4.2', '80805', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'February 26, 2018', 'Varies with device', '4.0.3 and up']


['Quick PDF Scanner + OCR FREE', 'BUSINESS', '4.2', '80805', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'February 26, 2018', 'Varies with device', '4.0.3 and up']


['Quick PDF Scanner + OCR FREE', 'BUSINESS', '4.2', '80804', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'February 26, 2018', 'Varies with device', '4.0.3 and up']


['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with devic

If we look at the Above Sample Data we can see that there are multiple entries(which is 1181) with various review numbers for a single App.  
if we take 'Instagram' for example, we can see that it alone has around 4 entries with difference in the review details.  
Inorder to remove the Redundant entry we need to choose a criterion and the best way to do it is by deleting various entries for the single app and keep only the most recent review ones which obviously is the ones having the highest number of reviews.  

In [7]:
reviews_max={}
for value_apps in Android_Apps[1:]:
    name=value_apps[0]
    n_reviews=value_apps[3]
    if name in reviews_max and reviews_max[name]< n_reviews:
        reviews_max[name]=n_reviews
    elif name not in reviews_max:
        reviews_max[name]=n_reviews
print(len(reviews_max))

9659


In [8]:
android_clean=[]     #stores our new cleaned data
already_added=[]     #stores the App name
for app in Android_Apps[1:]:
    name=app[0]
    n_reviews=app[3]
    if (reviews_max[name]==n_reviews) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)
        
explore_data(Android_Apps,2,5)
print(len(already_added))

['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


9659


We build only Apps focusing on the English Language, if we look at our Data set we can see that the Apps are also in other Languages so it' better we remove these.  
So we'll do this in the following:  

In [9]:
def eng_lang(string):
    count=0
    for i in string:
        
        if (ord(i)>127):
            count+=1
    if count>3:
        return False
    else:
        return True
eng_lang('Docs To Go™ Free Office Suite')

True

In [10]:
android_english = []
ios_english = []

for app in android_clean:
    name = app[0]
    if eng_lang(name):
        android_english.append(app)
        
for app in iOS_Apps[1:]:
    name = app[1]
    if eng_lang(name):
        ios_english.append(app)
        
explore_data(android_english, 0, 3, True)
print('\n')
explore_data(ios_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of Rows: 9614
Number of Columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

So far in the Data Cleaning Process, we have worked on :
1. Removing inaccurate data
2. Removing duplicate app entries
3. Removing non-english apps  
Next, we'll work on removing Apps which are Paid since our Company focuses on building only Free Apps.

In [11]:
iOS_free_apps=[]
android_free_apps=[]
for free_apps in ios_english[1:]:
    if free_apps[4]== '0.0':
        iOS_free_apps.append(free_apps)
for apps_free in android_english[1:]:
    if apps_free[7]== '0':
        android_free_apps.append(apps_free)
        
print (len(iOS_free_apps[0]))
print (len(android_free_apps))

16
8861


As mentioned earlier, our main aim is to determine the kind of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.  

To minimize the risk and overhead, our validation statergy for an app is comprised of three main steps:  

1. Build a minimal Android version of the App and add it to the Google Store.  
2. If there is a Good app response, we further develop it.
3.If the App is profitable after 6 months, we work on building the iOS version of the same.

Our end goal is to make apps for both the Google and Apple Stores, hence we need to find the app profiles that are successful in both the Markets.

Let's begin the analysis by getting a sense of the common genres for each market.

In [12]:
def freq_table(dataset,index):
    #dataset-list of lists, index-an integer type
    freq_table_list={}
    total=0
    for value in dataset[1:]:
        total+=1
        new_val=value[index]
        if new_val in freq_table_list:
            freq_table_list[new_val]+=1
        else:
            freq_table_list[new_val]=1
            
    percentage_val={}
    for value_per in freq_table_list:
        percentage=(freq_table_list[value_per]/total)*100.0
        percentage_val[value_per]=percentage
    return percentage_val

freq_table(iOS_free_apps,-5)



def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])
    

#print('\n')
#display_table(android_free_apps,1)

We can see that the Google play store has apps which are more diverse and well categorised but on the other hand it's not the same for Apple store.

Till now we can see that the apps created in iOS store is predominantly for fun while on the other hand the apps in Google store is more balanced, Let's Analyse further!  

In [13]:
display_table(iOS_free_apps,-5)

Games : 58.19875776397515
Entertainment : 7.888198757763975
Photo & Video : 4.937888198757764
Education : 3.6645962732919255
Social Networking : 3.260869565217391
Shopping : 2.608695652173913
Utilities : 2.515527950310559
Sports : 2.142857142857143
Music : 2.049689440993789
Health & Fitness : 2.018633540372671
Productivity : 1.7391304347826086
Lifestyle : 1.5838509316770186
News : 1.3354037267080745
Travel : 1.2422360248447204
Finance : 1.1180124223602486
Weather : 0.8695652173913043
Food & Drink : 0.8074534161490683
Reference : 0.5590062111801243
Business : 0.5279503105590062
Book : 0.43478260869565216
Navigation : 0.18633540372670807
Medical : 0.18633540372670807
Catalogs : 0.12422360248447205


In [14]:
value_apps=freq_table(iOS_free_apps,-5)
list_avg_user=[]
for genre in value_apps:
   
    total=0
    len_genre=0
    for value in iOS_free_apps:
        genre_App=value[-5]
        if genre_App == genre:
            rating_user=float(value[5])
            total+= rating_user
            len_genre+=1
    avg_user=(total/len_genre)
    list_avg_user.append(avg_user)
    print (genre,':',avg_user)
print(sorted(list_avg_user,reverse=True))

Social Networking : 43899.514285714286
News : 21248.023255813954
Food & Drink : 33333.92307692308
Music : 57326.530303030304
Shopping : 26919.690476190477
Travel : 28243.8
Education : 7003.983050847458
Reference : 74942.11111111111
Entertainment : 14029.830708661417
Sports : 23008.898550724636
Finance : 31467.944444444445
Book : 39758.5
Health & Fitness : 23298.015384615384
Medical : 612.0
Photo & Video : 28441.54375
Navigation : 86090.33333333333
Utilities : 18684.456790123455
Lifestyle : 16485.764705882353
Catalogs : 4004.0
Business : 7491.117647058823
Games : 22788.6696905016
Productivity : 21028.410714285714
Weather : 52279.892857142855
[86090.33333333333, 74942.11111111111, 57326.530303030304, 52279.892857142855, 43899.514285714286, 39758.5, 33333.92307692308, 31467.944444444445, 28441.54375, 28243.8, 26919.690476190477, 23298.015384615384, 23008.898550724636, 22788.6696905016, 21248.023255813954, 21028.410714285714, 18684.456790123455, 16485.764705882353, 14029.830708661417, 7491

On an average, Navigation Apps seem to have the highest User ratings amongst the others. but however, this value is bit skewed by Google and Waze as seen below:  

In [15]:
for letsnavi in iOS_free_apps:
    if letsnavi[-5]=='Navigation':
        print(letsnavi[1],':',letsnavi[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


Now we look at the Google App store's install column to check the number for our Analysis.  

In [16]:
values_appstore=freq_table(android_free_apps,1)
list_app_high=[]
for category in values_appstore:
    total=0
    len_category=0
    for cate in android_free_apps:
        category_app=cate[1]
        if category_app== category:
            len_category+=1
            n_install=cate[5]
            n_install=n_install.replace('+','')
            n_install=n_install.replace(',','')
            n_install=float(n_install)
            total+= n_install
    avg_instal=total/len_category
    print(category,':',avg_instal)
    list_app_high.append(avg_instal)
print(sorted(list_app_high,reverse=True))

SOCIAL : 23253652.127118643
EVENTS : 253542.22222222222
COMICS : 817657.2727272727
ART_AND_DESIGN : 2021626.7857142857
MEDICAL : 120616.48717948717
BOOKS_AND_REFERENCE : 8767811.894736841
LIFESTYLE : 1437816.2687861272
ENTERTAINMENT : 11640705.88235294
LIBRARIES_AND_DEMO : 638503.734939759
BUSINESS : 1712290.1474201474
BEAUTY : 513151.88679245283
NEWS_AND_MAGAZINES : 9549178.467741935
HEALTH_AND_FITNESS : 4188821.9853479853
FINANCE : 1387692.475609756
FAMILY : 3694276.334922527
PRODUCTIVITY : 16787331.344927534
PHOTOGRAPHY : 17805627.643678162
PERSONALIZATION : 5201482.6122448975
GAME : 15560965.599534342
WEATHER : 5074486.197183099
FOOD_AND_DRINK : 1924897.7363636363
MAPS_AND_NAVIGATION : 4056941.7741935486
SPORTS : 3638640.1428571427
EDUCATION : 1820673.076923077
COMMUNICATION : 38456119.167247385
VIDEO_PLAYERS : 24727872.452830188
TRAVEL_AND_LOCAL : 13984077.710144928
DATING : 854028.8303030303
HOUSE_AND_HOME : 1331540.5616438356
SHOPPING : 7036877.311557789
TOOLS : 10682301.0333778

In [17]:
print(len(iOS_free_apps))
print(len(android_free_apps))

3221
8861


On average, communication apps have the most installs: 38,456,119. This number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs:

In [18]:
for app in android_free_apps:
    if app[1]== 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

If we removed all the communication apps that have over 100 million installs, the average would be reduced roughly ten times:

In [19]:
under_100_m = []

for app in android_free_apps:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        under_100_m.append(float(n_installs))
        
sum(under_100_m) / len(under_100_m)

3603485.3884615386

We see the same pattern for the video players category, which is the runner-up with 24,727,872 installs. The market is dominated by apps like Youtube, Google Play Movies & TV, or MX Player. The pattern is repeated for social apps (where we have giants like Facebook, Instagram, Google+, etc.), photography apps (Google Photos and other popular photo editors), or productivity apps (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.).
Again, the main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.
The game genre seems pretty popular, but previously we found out this part of the market seems a bit saturated, so we'd like to come up with a different app recommendation if possible.
The books and reference genre looks fairly popular as well, with an average number of installs of 8,767,811. It's interesting to explore this in more depth, since we found this genre has some potential to work well on the App Store, and our aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play.
Let's take a look at some of the apps from this genre and their number of installs:

In [21]:
for app in android_free_apps:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E


The book and reference genre includes a variety of apps: software for processing and reading ebooks, various collections of libraries, dictionaries, tutorials on programming or languages, etc. It seems there's still a small number of extremely popular apps that skew the average:

In [23]:
for app in android_free_apps:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


However, it looks like there are only a few very popular apps, so this market still shows potential. Let's try to get some app ideas based on the kind of apps that are somewhere in the middle in terms of popularity (between 1,000,000 and 100,000,000 downloads):

In [24]:
for app in android_free_apps:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

This niche seems to be dominated by software for processing and reading ebooks, as well as various collections of libraries and dictionaries, so it's probably not a good idea to build similar apps since there'll be some significant competition.
We also notice there are quite a few apps built around the book Quran, which suggests that building an app around a popular book can be profitable. It seems that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets.
However, it looks like the market is already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.

### Conclusions
In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.
We concluded that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets. The markets are already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.